e2e/local_repository/local_repository_test.go

```diff
@@ -0,0 +1,105 @@
+package local_repository
+
+import (
+	"fmt"
+	// "os"
+	"runtime/debug"
+	"strings"
+	"testing"
+	"errors"
+
+	bazel "github.com/bazelbuild/bazel-integration-testing/go"
+	// "github.com/bazelbuild/bazel-watcher/e2e"
+)
+
+func must(t *testing.T, e error) {
+	if e != nil {
+		t.Fatalf("Error: %s", e)
+		t.Logf("Stack trace:\n%s", string(debug.Stack()))
+	}
+}
+
+func TestLocalRepositoryRunWithModifiedFile(t *testing.T) {
+	secondary, err := bazel.New()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	must(t, secondary.ScratchFile("WORKSPACE", `workspace(name = "secondary")`))
+	must(t, secondary.ScratchFile("BUILD", `
+sh_library(
+	name = "lib",
+	data = ["lib.sh"],
+	visibility = ["//visibility:public"],
+)
+
+sh_binary(
+	name = "workspace",
+	srcs = ["workspace.sh"],
+)
+`))
+	must(t, secondary.ScratchFileWithMode("lib.sh", `
+function say_hello {
+	printf "hello!"
+}
+`, 0777))
+	must(t, secondary.ScratchFileWithMode("workspace.sh", `
+echo $BUILD_WORKSPACE_DIRECTORY
+`, 0777))
+
+	_, stdout, stderr := secondary.RunBazel([]string{"run", "//:workspace"})
+	secondaryWorkspacePath := strings.TrimSpace(stdout)
+
+	main, err := bazel.New()
+	if err != nil {
+		t.Fatal(err)
+	}
+	must(t, main.ScratchFile("WORKSPACE", fmt.Sprintf(`
+workspace(name = "main")
+
+local_repository(
+    name = "secondary",
+    path = "%s",
+)
+`, secondaryWorkspacePath)))
+	must(t, main.ScratchFile("BUILD", `
+sh_binary(
+	name = "test",
+	srcs = ["test.sh"],
+	deps = [
+		"@secondary//:lib",
+	],
+)
+`))
+	must(t, main.ScratchFileWithMode("test.sh", `
+#!/bin/bash
+source ../secondary/lib.sh
+say_hello
+`, 0777))
+
+	_, stdout, stderr = main.RunBazel([]string{"run", "//:test"})
+
+	fmt.Print(stdout)
+	fmt.Print("-----\n")
+	fmt.Print(stderr)
+
+	t.Fatal(errors.New("exit"))
+
+// 	ibazel := e2e.NewIBazelTester(t, main)
+// 	ibazel.Run([]string{}, "//:test")
+// 	defer ibazel.Kill()
+
+// 	ibazel.ExpectOutput("hello!")
+
+// 	// File operations in `TestingBazel` doesn't respect their own `tmpDir` all instances
```

Comment:
The problem is [here](https://github.com/bazelbuild/bazel-integration-testing/blob/master/go/bazel.go#L58). After creating new `TestingBazel` instance all file operations in other instances are made in the context of last created instance because `Scratch*` methods rely on the current `pwd` and not `tmpDir`.